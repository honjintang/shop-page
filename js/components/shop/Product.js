import React from 'react'
import Card from 'react-bootstrap/Card'
import heart from '../../../assets/images/heart-outline.svg'

const Product = ({id, img, title, brand, price, size, sold }) => {
    return (
        <Card id={id}>
            <Card.Img className={sold ? 'card-img-sold' : null} variant="top" src={img} />
            {sold && <Card.ImgOverlay>
                <Card.Title>SOLD</Card.Title>
            </Card.ImgOverlay>}
            {<Card.ImgOverlay>
                <Card.Title className='like'>
                    <img className='heart' src={heart} width="22.5px" height="22.5px"/>
                </Card.Title>
            </Card.ImgOverlay>}
            <Card.Body>
            <Card.Title>{title}</Card.Title>
            <Card.Text>
                <p>Brand: {brand || 'N/A'}</p>
                <p>Size: {size}</p>
                <p><b>Price: Â£{price}</b></p>
            </Card.Text>
            </Card.Body>
        </Card>
    )
} 

export default Product