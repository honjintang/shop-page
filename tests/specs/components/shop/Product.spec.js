import React from 'react'
import { shallow } from 'enzyme'
import Product from '~/components/shop/Product'

describe('components/shop/Product', () => {
    let component

    describe('matches snapshot', () => {
        it('renders image with title, brand, size and price', () => {
            component = shallow(<Product 
                id='mockId'
                title='mockTitle'
                brand='mockBrand'
                size='mockSize'
                price='mockPrice'
                sold={false}
                 />)
            expect(component).toMatchSnapshot()
        })

        it('when product is sold renders with overlay', () => {
            component = shallow(<Product 
                id='mockId'
                title='mockTitle'
                size='mockSize'
                price='mockPrice'
                sold={true}
                 />)
            expect(component).toMatchSnapshot()
        })

        it('when brand is not present renders default N/A', () => {
            component = shallow(<Product 
                id='mockId'
                title='mockTitle'
                size='mockSize'
                price='mockPrice'
                sold={false}
                 />)
            expect(component).toMatchSnapshot()
        })
    })
})