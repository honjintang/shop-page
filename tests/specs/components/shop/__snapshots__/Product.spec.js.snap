// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/shop/Product matches snapshot renders image with title, brand, size and price 1`] = `
<Card
  body={false}
  id="mockId"
>
  <CardImg
    className={null}
    variant="top"
  />
  <CardImgOverlay>
    <CardTitle
      className="like"
    >
      <img
        className="heart"
        height="22.5px"
        src="test-file-stub"
        width="22.5px"
      />
    </CardTitle>
  </CardImgOverlay>
  <CardBody>
    <CardTitle>
      mockTitle
    </CardTitle>
    <CardText>
      <p>
        Brand: 
        mockBrand
      </p>
      <p>
        Size: 
        mockSize
      </p>
      <p>
        <b>
          Price: £
          mockPrice
        </b>
      </p>
    </CardText>
  </CardBody>
</Card>
`;

exports[`components/shop/Product matches snapshot when brand is not present renders default N/A 1`] = `
<Card
  body={false}
  id="mockId"
>
  <CardImg
    className={null}
    variant="top"
  />
  <CardImgOverlay>
    <CardTitle
      className="like"
    >
      <img
        className="heart"
        height="22.5px"
        src="test-file-stub"
        width="22.5px"
      />
    </CardTitle>
  </CardImgOverlay>
  <CardBody>
    <CardTitle>
      mockTitle
    </CardTitle>
    <CardText>
      <p>
        Brand: 
        N/A
      </p>
      <p>
        Size: 
        mockSize
      </p>
      <p>
        <b>
          Price: £
          mockPrice
        </b>
      </p>
    </CardText>
  </CardBody>
</Card>
`;

exports[`components/shop/Product matches snapshot when product is sold renders with overlay 1`] = `
<Card
  body={false}
  id="mockId"
>
  <CardImg
    className="card-img-sold"
    variant="top"
  />
  <CardImgOverlay>
    <CardTitle>
      SOLD
    </CardTitle>
  </CardImgOverlay>
  <CardImgOverlay>
    <CardTitle
      className="like"
    >
      <img
        className="heart"
        height="22.5px"
        src="test-file-stub"
        width="22.5px"
      />
    </CardTitle>
  </CardImgOverlay>
  <CardBody>
    <CardTitle>
      mockTitle
    </CardTitle>
    <CardText>
      <p>
        Brand: 
        N/A
      </p>
      <p>
        Size: 
        mockSize
      </p>
      <p>
        <b>
          Price: £
          mockPrice
        </b>
      </p>
    </CardText>
  </CardBody>
</Card>
`;
